flare/kernels/utils.py

```diff
@@ -98,19 +106,37 @@ def from_mask_to_args(hyps, hyps_mask: dict, cutoffs):
     else:
         orig_hyps = hyps
 
-    if (n2b != 0):
-        sig2 = orig_hyps[:n2b]
-        ls2 = orig_hyps[n2b:n2b * 2]
-    if (n3b !=0):
-        sig3 = orig_hyps[n2b * 2:n2b * 2 + n3b]
-        ls3 = orig_hyps[n2b * 2 + n3b:n2b * 2 + n3b * 2]
-    if (n2b == 0) and (n3b == 0):
-        raise NameError("Hyperparameter mask missing nbond and/or"
-                        "ntriplet key")
-
-    return (np.array(cutoffs), hyps_mask['nspec'], np.array(hyps_mask['spec_mask']),
-            n2b, np.array(bond_mask), n3b, np.array(triplet_mask),
-            np.array(sig2), np.array(ls2), np.array(sig3), np.array(ls3))
+    if (len(cutoffs)<=2):
+        if (n2b != 0):
+            sig2 = orig_hyps[:n2b]
+            ls2 = orig_hyps[n2b:n2b * 2]
+        if (n3b !=0):
+            sig3 = orig_hyps[n2b * 2:n2b * 2 + n3b]
+            ls3 = orig_hyps[n2b * 2 + n3b:n2b * 2 + n3b * 2]
+        if (n2b == 0) and (n3b == 0):
+            raise NameError("Hyperparameter mask missing nbond and/or"
+                            "ntriplet key")
+
+        return (np.array(cutoffs), hyps_mask['nspec'], np.array(hyps_mask['spec_mask']),
+                n2b, np.array(bond_mask), n3b, np.array(triplet_mask),
+                np.array(sig2), np.array(ls2), np.array(sig3), np.array(ls3))
+    elif (len(cutoffs)==3):
+        sigm = None
+        lsm = None
```

Comment:
these should probably be moved outside of the if statement

Reply:
ok. just changed it.